crates/kitsune_p2p/proxy/src/config.rs

```diff
@@ -0,0 +1,156 @@
+use crate::*;
+
+/// Callback function signature for proxy accept/deny.
+pub type AcceptProxyCallbackFn =
+    Arc<dyn Fn(CertDigest) -> MustBoxFuture<'static, bool> + 'static + Send + Sync>;
+
+/// Callback type for proxy accept/deny.
+#[derive(Clone, Deref, AsRef)]
+pub struct AcceptProxyCallback(pub AcceptProxyCallbackFn);
+
+impl AcceptProxyCallback {
+    /// Callback that blanket denies all proxy requests.
+    pub fn reject_all() -> Self {
+        Self(Arc::new(|_| async { false }.boxed().into()))
+    }
+
+    /// Callback that blanket accepts all proxy requests.
+    pub fn accept_all() -> Self {
+        Self(Arc::new(|_| async { true }.boxed().into()))
+    }
+}
+
+/// Tls Configuration for proxy.
+#[derive(Clone)]
+pub struct TlsConfig {
+    /// Cert
+    pub cert: Cert,
+
+    /// Cert Priv Key
+    pub cert_priv_key: CertPrivKey,
+
+    /// Cert Digest
+    pub cert_digest: CertDigest,
+}
+
+impl TlsConfig {
+    /// Create a new ephemeral tls certificate that will not be persisted.
+    pub async fn new_ephemeral() -> TransportResult<Self> {
+        let mut options = lair_keystore_api::actor::TlsCertOptions::default();
+        options.alg = lair_keystore_api::actor::TlsCertAlg::PkcsEcdsaP256Sha256;
+        let cert = lair_keystore_api::internal::tls::tls_cert_self_signed_new_from_entropy(options)
+            .await
+            .map_err(TransportError::other)?;
+        Ok(Self {
+            cert: cert.cert_der,
+            cert_priv_key: cert.priv_key_der,
+            cert_digest: cert.cert_digest,
+        })
+    }
+}
+
+/// Configuration for proxy binding.
+pub enum ProxyConfig {
+    /// We want to be hosted at a remote proxy location.
+    RemoteProxyClient {
```

Comment:
So this allows another proxy to forward traffic to and from use?

Reply:
Yeah, exactly - we make the outgoing connection to the proxy - then anyone else talking to that proxy can establish channels that route all the way down through our connection to us.